################################################################################
SHORT DESCRIPTION: 
################################################################################
The xebfb2 framebuffer driver in UIVM in XT relies on being able to set caching
attributes on pages, through PAT.

################################################################################
LONG DESCRIPTION: 
################################################################################
But, the disallow mask for pv domains does not allow them unless the PV domain
is using passthrough devices. Just allow them for all pv domains as well. We 
had existing patch for this in XT, but it was uglier as it manipulated the check
call sites instead of disallow mask, as well as it had confusing name
"hvm_get_mem_pinned_cacheattr_always"

################################################################################
CHANGELOG 
################################################################################
Author: Tomasz Wroblewski <tomasz.wroblewski@citrix.com>

################################################################################
REMOVAL 
################################################################################

################################################################################
UPSTREAM PLAN
################################################################################
None.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################

################################################################################
PATCHES 
################################################################################
--- a/xen/arch/x86/mm.c
+++ b/xen/arch/x86/mm.c
@@ -164,13 +164,7 @@ static uint32_t base_disallow_mask;
 
 #define L4_DISALLOW_MASK (base_disallow_mask)
 
-#define l1_disallow_mask(d)                                     \
-    ((d != dom_io) &&                                           \
-     (rangeset_is_empty((d)->iomem_caps) &&                     \
-      rangeset_is_empty((d)->arch.ioport_caps) &&               \
-      !has_arch_pdevs(d) &&                                     \
-      is_pv_domain(d)) ?                                        \
-     L1_DISALLOW_MASK : (L1_DISALLOW_MASK & ~PAGE_CACHE_ATTRS))
+#define l1_disallow_mask(d) (L1_DISALLOW_MASK & ~PAGE_CACHE_ATTRS)
 
 static s8 __read_mostly opt_mmio_relax;
 
